block/manager.go

```diff
@@ -328,3 +330,22 @@ func (m *Manager) setDA(daconfig string, dalcKV store.KV, logger log.Logger) err
 	m.Retriever = retriever
 	return nil
 }
+
+func (m *Manager) gossipPendingBlocks(ctx context.Context) {
+	for h := m.State.LastGossipedHeight + 1; h <= m.State.Height(); h++ {
```

Comment:
after upgrade the LastGossipedHiehgt will be zero no? so it will gossip all blocks.

Reply:
true. i set it to the node height in case its 0 in the init. this way it will not start from zero when the upgrade.